pytube/extract.py

```diff
@@ -269,31 +271,29 @@ def get_ytplayer_config(html: str) -> Any:
     """
     logger.debug("finding initial function name")
     config_patterns = [
-        r"ytplayer\.config\s*=\s*({.+?});ytplayer",
-        r"ytInitialPlayerResponse\s*=\s*({.+?(?<!gdpr)});"
+        r"ytplayer\.config\s*=\s*",
```

Comment:
usually ? after *  to choose the shortest match or it can get really really long if an extra character gets added

Reply:
I would agree if we were matching with `\.*`, but we're just matching whitespace, and in the case of the second `\s*`, we have to match *all* whitespace preceding the `{` character. In the actual string we're searching, to the best of my knowledge, this always appears in the text as `ytplayer.config = {<stuff here>}`. If we were to use, say `ytplayer\.config\s*?=\s*?`, the space preceding the actual object would not be captured, and `parse_for_object` would fail.